### PR TITLE
Fix HDFSHook HAClient is invalid

### PR DESCRIPTION
**Pull request by trickysky**
BaseHook.get_connections is deprecated, only return one connection
snakebite HAClient need more than one connection 

[HAClient](https://github.com/apache/airflow/blob/main/airflow/providers/apache/hdfs/hooks/hdfs.py#L98)

### CODE REVIEW

**Comment by trickysky** [comment]
Another code implementation:

**Comment by potiuk** [comment]
I think we need a unit test covering it to prevent regressions. It's suspicious not see tests modified/added/fixed with such a change

**Comment by trickysky** [comment]
> I think we need a unit test covering it to prevent regressions. It's suspicious not see tests modified/added/fixed with such a change

I modify a bad unit test, please take a look, thanks

File: tests/providers/apache/hdfs/hooks/test_hdfs.py [hunk 1]
```diff
@@ -74,10 +74,13 @@ def test_get_autoconfig_client_no_conn(self, mock_client):
         HDFSHook(hdfs_conn_id="hdfs_missing", autoconfig=True).get_conn()
         mock_client.assert_called_once_with(effective_user=None, use_sasl=False)
 
-    @mock.patch("airflow.providers.apache.hdfs.hooks.hdfs.HDFSHook.get_connections")
-    def test_get_ha_client(self, mock_get_connections):
-        conn_1 = Connection(conn_id="hdfs_default", conn_type="hdfs", host="localhost", port=8020)
-        conn_2 = Connection(conn_id="hdfs_default", conn_type="hdfs", host="localhost2", port=8020)
-        mock_get_connections.return_value = [conn_1, conn_2]
```
Comment (trickysky) [inline tests/providers/apache/hdfs/hooks/test_hdfs.py:81]: Connection model `conn_id` field is unique

**Review by potiuk — APPROVED** [review]

**Pull request merged**